cmd/kubeadm/app/util/version.go

```diff
@@ -24,17 +24,20 @@ import (
 	"strings"
 	"time"
 
+	"github.com/golang/glog"
 	netutil "k8s.io/apimachinery/pkg/util/net"
+	versionutil "k8s.io/kubernetes/pkg/util/version"
+	"k8s.io/kubernetes/pkg/version"
 )
 
 const (
-	getReleaseVersionTimeout = time.Duration(10 * time.Second)
+	getReleaseVersionTimeout = time.Duration(4 * time.Second)
 )
 
 var (
 	kubeReleaseBucketURL  = "https://dl.k8s.io"
 	kubeReleaseRegex      = regexp.MustCompile(`^v?(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)([-0-9a-zA-Z_\.+]*)?$`)
-	kubeReleaseLabelRegex = regexp.MustCompile(`^[[:lower:]]+(-[-\w_\.]+)?$`)
+	kubeReleaseLabelRegex = regexp.MustCompile(`^(latest|stable)(-([0-9]+)(\.([0-9]+))?)?$`)
```

Comment:
I don't see need to change that regexp. Especially to put it so strict with only `latest` or `stable`. We don't want to modify that piece of code every time when somebody creates new label on release server.

Reply:
so there is a problem here with that and the fallback.

this integration test fails without the regexp local validation change:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/test/cmd/init_test.go#L92-L94

the reason is that when `foo` is passed the version is checked from the internet and when it fails (as there is no `foo` label) this falls back to the local client version and it succeeds. the 404 handling proposal should fix that.

Reply:
I think it will be better to fix the test case linked above. This particular test case will be failing on `foobar` value based on absence on the internet or in case of network problems. So, it will be a lot better to have validation if it passes something that is clearly as incorrect semver (e.g. `1.2`) rather than something that looks like label and potentially trigger network operations during the test execution.
 
For validation of those regexps and network functions, we have separate test cases:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/version_test.go#L100
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/version_test.go#L82

So, I'd prefer to change value in https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/test/cmd/init_test.go#L92-L94 than modifying logic and regexps in `cmd/kubeadm/app/util/version.go`